### PR TITLE
[REVIEW] Add cuPy as a formal cuDF dependency

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Adds `cuPy` as a formal dependency to `cuDF`.

### CODE REVIEW

File: conda/environments/cudf_dev_cuda10.0.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ channels:
   - conda-forge
   - defaults
 dependencies:
+  - rapidsai/label/cuda10.0::cupy>=6*
```
Comment (kkraus14) [inline conda/environments/cudf_dev_cuda10.0.yml]: Shouldn't need to specify the channel here, the labels will handle it for us.

File: conda/environments/cudf_dev_cuda10.0.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ channels:
...
   - rmm=0.10.*
```
Comment (kkraus14) [inline conda/environments/cudf_dev_cuda10.0.yml]: Can you bump this to 0.11 in the PR as well?

File: conda/environments/cudf_dev_cuda9.2.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ channels:
   - conda-forge
   - defaults
 dependencies:
+  - rapidsai/label/cuda9.2::cupy>=6*
```
Comment (kkraus14) [inline conda/environments/cudf_dev_cuda9.2.yml]: Same here

**Comment by brandon-b-miller** [comment]
per a slack conversation with @kkraus14 we'll leave in the optional `cuPy` code for one release so as not to potentially perturb downstream devs.

File: conda/environments/cudf_dev_cuda10.0.yml (hunk 1 above)
Comment (brandon-b-miller) [reply conda/environments/cudf_dev_cuda10.0.yml]: updated

File: conda/environments/cudf_dev_cuda10.0.yml (hunk 2 above)
Comment (brandon-b-miller) [reply conda/environments/cudf_dev_cuda10.0.yml]: done

File: conda/environments/cudf_dev_cuda9.2.yml (hunk 3 above)
Comment (brandon-b-miller) [reply conda/environments/cudf_dev_cuda9.2.yml]: updated

**Comment by brandon-b-miller** [comment]
done, assuming the only thing that is different is the cuda toolkit version

**Pull request merged**